### PR TITLE
Adding a helper subscriber

### PR DESCRIPTION
**Pull request by akshaisarma**

### CODE REVIEW

File: src/test/java/com/yahoo/bullet/pubsub/BufferingSubscriberTest.java [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+            callCount++;
+            if (testMessages == null || testMessages.isEmpty()) {
+                return testMessages;
+            }
+            return Collections.singletonList(testMessages.remove(0));
+        }
+
+        @Override
+        public void close() {
+            testMessages = new LinkedList<>();
```
Comment (hbxie) [inline src/test/java/com/yahoo/bullet/pubsub/BufferingSubscriberTest.java]: testMessages.clear() ?

Comment (akshaisarma) [reply src/test/java/com/yahoo/bullet/pubsub/BufferingSubscriberTest.java]: On purpose to test what haveMessages does when we return an empty list. See line 84 in this file

File: src/main/java/com/yahoo/bullet/pubsub/BufferingSubscriber.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This implements a {@link Subscriber} that provides a base subscriber that buffers a fixed number of messages read.
+ * See {@link BufferingSubscriber#maxUncommitedMessages}.
+ *
+ * It provides  implementations of {@link Subscriber#commit(String, int)}, {@link Subscriber#fail(String, int)} and
```
[the remark this answers is not in the document]
Comment (akshaisarma) [reply src/main/java/com/yahoo/bullet/pubsub/BufferingSubscriber.java]: Done

File: src/test/java/com/yahoo/bullet/pubsub/BufferingSubscriberTest.java (hunk 1 above)
Comment (akshaisarma) [reply src/test/java/com/yahoo/bullet/pubsub/BufferingSubscriberTest.java]: Sorry misread your comment. Will fix.

**Pull request merged**